### PR TITLE
feat: "create team" flow UI [AR-848]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-848" title="AR-848" target="_blank">SubtaskAR-848</a>  implement the registration team flow UI
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

User needs to be able to fill out a personal account creation flow as well as team creation flow. Both flows share part of UI and logic so it doesn’t make sense to implement and maintain both flows separately - common part of both can be reused. There were also some smaller things in the creation flow that needed to be fixed or improved (listed below).

### Solutions

- implemented screen, viewmodel and navigation items for the “create team” flow
- logic common to both flows separated into CreateAccountBaseViewModel
- added transition animation in creation flows
- added CreateAccountFlowType to steps states which is taken from base viewmodel instead of direct screen parameters (title, subtitle, etc.)
- added “team name” input field to “details” screen for the “create team” flow
- added “loading” state (circular progress indicator) to CreateAccountCodeScreen
- fixed keyboard not disappearing after entering full code and when navigating between NavigationItems
- fixed “disabled” state in CodeTextField
- WireCircularProgressIndicator to use compose CircularProgressIndicator, which is already available in the version we are using, instead of AndroidView and view-based one

### Attachments (Optional)

https://user-images.githubusercontent.com/30429749/158341049-5a3e9789-fb14-4b5f-93b4-9628f8465b68.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by saleniuk** [comment]
half of the changes are added resources 🙈

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import com.wire.android.ui.authentication.create.details.CreateAccountDetailsViewModel
+import com.wire.android.ui.authentication.create.details.CreateAccountDetailsViewState
+import com.wire.android.ui.authentication.create.email.CreateAccountEmailViewModel
+import com.wire.android.ui.authentication.create.email.CreateAccountEmailViewState
+import com.wire.android.ui.authentication.create.overview.CreateAccountOverviewViewModel
+import com.wire.android.ui.common.textfield.CodeFieldValue
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+@Suppress("MagicNumber", "TooManyFunctions")
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt:25]: Can we add a TODO to remove the MagicNumber once the logic is in place?

Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt]: 🚀 Actually it's not needed anymore, I duplicated it mindlessly, I'll pay more attention to it

**Comment by saleniuk** [comment]
I also made an update of the flows according to the recent Wolfgang's design changes (move "set username" step after "summary" and make it more universal and not a part of the creation flow)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <string name="label_cancel">Cancel</string>
     <string name="label_confirm">Confirm</string>
     <string name="label_continue">Continue</string>
+    <string name="label_confirm">Confirm</string>
```
Comment (gongracr) [inline app/src/main/res/values/strings.xml:11]: This string was duplicated...

**Pull request merged**